gateway/mw_access_rights.go

```diff
@@ -25,38 +25,32 @@ func (a *AccessRightsCheck) ProcessRequest(w http.ResponseWriter, r *http.Reques
 	session := ctxGetSession(r)
 
 	// If there's nothing in our profile, we let them through to the next phase
-	if len(session.AccessRights) > 0 {
-		// Otherwise, run auth checks
-		versionList, apiExists := session.AccessRights[a.Spec.APIID]
-		if !apiExists {
-			a.Logger().Info("Attempted access to unauthorised API")
-			return errors.New("Access to this API has been disallowed"), http.StatusForbidden
-		}
+	if len(session.AccessRights) == 0 {
+		return nil, http.StatusOK
+	}
 
-		// Find the version in their key access details
-		found := false
-		if a.Spec.VersionData.NotVersioned {
-			// Not versioned, no point checking version access rights
-			found = true
-		} else {
-			targetVersion := a.Spec.getVersionFromRequest(r)
-			if targetVersion == "" {
-				targetVersion = a.Spec.VersionData.DefaultVersion
-			}
-
-			for _, vInfo := range versionList.Versions {
-				if vInfo == targetVersion {
-					found = true
-					break
-				}
-			}
-		}
+	// Otherwise, run auth checks
+	versionList, apiExists := session.AccessRights[a.Spec.APIID]
+	if !apiExists {
+		a.Logger().Info("Attempted access to unauthorised API")
+		return errors.New("Access to this API has been disallowed"), http.StatusForbidden
+	}
+
+	if a.Spec.VersionData.NotVersioned {
+		return nil, http.StatusOK
+	}
+
+	targetVersion := a.Spec.getVersionFromRequest(r)
+	if targetVersion == "" {
+		targetVersion = a.Spec.VersionData.DefaultVersion
```

Comment:
Should we move this to the end of the function ; if it's set it should be the fallback?

Reply:
but it is in the search below in the `for` loop.